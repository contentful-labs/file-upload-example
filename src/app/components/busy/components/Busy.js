import { h, Component } from 'preact'
import proptypes from 'proptypes'

import LoadingIndicator from 'components/loading-indicator'

export default class Busy extends Component {
  static propTypes = {
    isBusy: proptypes.bool.isRequired,
    message: proptypes.string
  }
  render ({ isBusy, message }) {
    if (!isBusy) {
      return null
    }
    return (
      <div>
        <LoadingIndicator message />
      </div>
    )
  }
}
