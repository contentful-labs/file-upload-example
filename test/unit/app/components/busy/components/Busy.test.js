import { h } from 'preact'
import render from 'preact-render-to-string'

import Busy from '../../../../../../src/app/components/busy/components/Busy'

test('Busy - is busy', () => {
  const html = render(<Busy isBusy message={'doing some heavy lifting over here'} />)
  expect(html).toMatchSnapshot()
})

test('Busy - not busy', () => {
  const html = render(<Busy isBusy={false} />)
  expect(html).toMatchSnapshot()
})
