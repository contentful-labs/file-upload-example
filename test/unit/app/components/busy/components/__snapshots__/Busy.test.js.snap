// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Busy - is busy 1`] = `
<div>
  <div>
    <div class="wrapper">
      <div class="content">
        <div class="logo">
          <img src="test-file-stub" alt="Contentful" />
        </div>
        <div class="message">doing some heavy lifting over here</div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Busy - not busy 1`] = `""`;
